### PR TITLE
feat: add a default syscall_config.yaml file

### PR DESCRIPTION
**Pull request by JPLachance**
## Summary

Lacework now supports syscall monitoring. From the work done in https://github.com/lacework/terraform-aws-ssm-agent/pull/46/files, I created the required syscall_config.yaml file in the EKS DaemonSet.

## How did you test this change?

The change got tested in our development environment.

## Issue

A new syscall_config.yaml file is required by the Lacework agent. This is a naive first implementation.

### CODE REVIEW

**Comment by afiune** [comment]
Make it so

File: lacework_node.tf [hunk 1]
```diff
@@ -48,7 +49,8 @@ resource "kubernetes_secret" "lacework_config" {
   }
 
   data = {
-    "config.json" = local.merged_node_config
+    "config.json"         = local.merged_node_config
+    "syscall_config.yaml" = local.node_syscall_config_data
```
Comment (nschmeller) [inline lacework_node.tf:53]: Could you add a new Terraform variable to toggle syscall enabled / disabled and gate the inclusion of the syscall config with this toggle?

File: lacework_node.tf [hunk 2]
```diff
@@ -268,6 +270,10 @@ resource "kubernetes_daemonset" "lacework_datacollector" {
               key  = "config.json"
               path = "config.json"
             }
+            items {
```
Comment (nschmeller) [inline lacework_node.tf:273]: Could you add a new Terraform variable to toggle syscall enabled / disabled and gate the inclusion of the syscall config with this toggle?

**Review by nschmeller — CHANGES_REQUESTED** [review]
Hi @JPLachance, this looks super cool, thanks for submitting a PR!

I have one request described in the review comments -- could we also include a way to enable / disable adding the syscall config file? Not all of our customers want syscall tracking just yet :)

Thanks!!

**Review by nschmeller — COMMENTED** [review]
What do you think about using https://developer.hashicorp.com/terraform/language/expressions/conditionals? I left some snippets with how I might do that. It's not very elegant, so if you think of a better way, feel free to do that!

File: lacework_node.tf (hunk 2 above)
Comment (JPLachance) [reply lacework_node.tf]: After many many tentatives, I found a way to make it work using an array and a dynamic block.

File: lacework_node.tf (hunk 1 above)
Comment (JPLachance) [reply lacework_node.tf:56]: Done!

File: examples/disable-stdout-logging/main.tf [hunk 3]
```diff
@@ -10,6 +10,6 @@ resource "lacework_agent_access_token" "k8s" {
 module "lacework_k8s_datacollector" {
   source = "../../"
 
-  lacework_access_token    = lacework_agent_access_token.k8s.token
+  lacework_access_token     = lacework_agent_access_token.k8s.token
```
Comment (nschmeller) [inline examples/disable-stdout-logging/main.tf:13]: Could you remove this change from the PR?

File: examples/custom-proxy-url/main.tf [hunk 4]
```diff
@@ -4,5 +4,5 @@ module "lacework_k8s_datacollector" {
   source = "../../"
 
   lacework_access_token = "0123456789ABCDEF0123456789ABCDEF"
-  lacework_proxy_url   = "https://proxy.address.here:8080"
+  lacework_proxy_url    = "https://proxy.address.here:8080"
```
Comment (nschmeller) [inline examples/custom-proxy-url/main.tf:7]: Could you remove this change from the PR?

File: examples/custom-agent-configuration/main.tf [hunk 5]
```diff
@@ -9,9 +9,9 @@ module "lacework_k8s_datacollector" {
       "capsmode" : "enabled"
     }
     "fim" : {
-      "runat": "23:50"
-      "mode": "enable"
+      "runat" : "23:50"
```
Comment (nschmeller) [inline examples/custom-agent-configuration/main.tf:12]: Could you remove these changes from the PR?

**Review by nschmeller — COMMENTED** [review]
Super cool!!!! TIL about the `dynamic` keyword 😀

I have two more suggestions:

1. It looks like there were some unrelated files that got caught in a `terraform fmt` -- would you mind keeping them separate from the PR?
2. Could you update the toplevel README to update https://github.com/lacework/terraform-kubernetes-agent#inputs? This will update the Terraform registry documentation. An example to do this is https://github.com/lacework/terraform-aws-ssm-agent/pull/50

This looks super cool, thanks again for taking the time!!!

File: examples/disable-stdout-logging/main.tf (hunk 3 above)
Comment (JPLachance) [reply examples/disable-stdout-logging/main.tf:13]: https://github.com/lacework/terraform-kubernetes-agent/pull/72

Merge that one, then I'll rebase or something.

File: examples/custom-proxy-url/main.tf (hunk 4 above)
Comment (JPLachance) [reply examples/custom-proxy-url/main.tf:7]: https://github.com/lacework/terraform-kubernetes-agent/pull/72

Merge that one, then I'll rebase or something.

File: examples/custom-agent-configuration/main.tf (hunk 5 above)
Comment (JPLachance) [reply examples/custom-agent-configuration/main.tf:12]: https://github.com/lacework/terraform-kubernetes-agent/pull/72

Merge that one, then I'll rebase or something.

**Comment by JPLachance** [comment]
Closing in favour of https://github.com/lacework/terraform-kubernetes-agent/pull/73 since I made a mess in the commits history.

**Pull request closed without merging**